source/assert.c

```diff
@@ -18,11 +18,24 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+static assert_handler_fn_t at_assert = NULL;
+
 void aws_debug_break(void);
 
 void aws_fatal_assert(const char *cond_str, const char *file, int line) {
     aws_debug_break();
-    fprintf(stderr, "Fatal error condition occurred in %s:%d: %s\nExiting Application\n", file, line, cond_str);
-    aws_backtrace_print(stderr, NULL);
-    abort();
+
+    if (at_assert) {
+        at_assert(cond_str, file, line);
+    } else {
```

Comment:
I could also just make this a default handler, and make it always call whatever is set as the handler.

Reply:
That will reduce branches, maybe worth doing? I would:
* Make the default assert handler a publicly exported function, in case someone else wants to call it.
* Finally add that `aws_common_library_init()` function we've been meaning to add in forever, and install it there.